File: fixtures/scale/ruby/large/lib/check_linearmix.rb
require_relative 'linearmix_encrypt'
require_relative 'linearmix_decrypt'
require_relative 'corpus_affine'

def check_linearmix
  corpus_affine.each do |text|
    sealed = linearmix_encrypt(text, 15)
    opened = linearmix_decrypt(sealed, 15)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_linearmix.py
from cipherlab.ciphers.linearmix import linearmix_encrypt, linearmix_decrypt
from cipherlab.specs.linearmix_spec import linearmix_spec
from cipherlab.corpus.corpus_affine import corpus_affine


def check_linearmix():
    spec = linearmix_spec()
    for text in corpus_affine():
        sealed = linearmix_encrypt(text, spec["key"])
        opened = linearmix_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "affine"
